samcli/lib/config/samconfig.py

```diff
@@ -39,11 +40,16 @@ def __init__(self, config_dir, filename=None):
             Optional. Name of the configuration file. It is recommended to stick with default so in the future we
             could automatically support auto-resolving multiple config files within same directory.
         """
+        self.document = {}
         self.filepath = Path(config_dir, filename or DEFAULT_CONFIG_FILE_NAME)
+        self.file_manager = self.FILE_MANAGER_MAPPER.get(self.filepath.suffix[1:], None)  # default to TOML
+        if not self.file_manager:
+            LOG.warning(f"The config file extension '{self.filepath.suffix[1:]}' is not supported. TOML will be used.")
```

Comment:
Open question: should we explicitly require the file format instead of assuming TOML by default?

Reply:
I personally think this should be changed to explicitly require a file extension. If I was to pass in a file format (such as `myconfig.xml`) that isn't supported, it'll try to parse it as TOML. The first part of that message is clear in reasoning/intent (not supported), but then from a UX stand point, it wouldn't quite make sense to try to parse it as TOML anyways.

Reply:
Yup, as per our conversation on Friday, being explicit is probably better in this situation. I'll have it implemented in the next commit 👍